### PR TITLE
victor-mono: init at 1.2.1

### LINKED ISSUE #64465: Font Request: Victor Mono
[Victor Mono](https://rubjo.github.io/victor-mono/) is a free programming font with semi-connected cursive italics and symbolic ligatures. 

The typeface is clean, crisp and narrow, with a large x-height. It is optimised for legible code, and comes in seven weights. It is available in Roman, Italic and Oblique styles.

image

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Fixes #64465.

Haven't tested installation but inspecting output LGTM-- requester want
to confirm this works for you? :)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/data/fonts/victor-mono/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
+{ lib, fetchzip }:
+
+let
+  version = "1.2.1";
+in fetchzip rec {
+  name = "victor-mono-${version}";
+
+  url = "https://github.com/rubjo/victor-mono/archive/v1.2.1.zip";
```
Comment (jonringer) [inline pkgs/data/fonts/victor-mono/default.nix]: be easier to update
```suggestion
  url = "https://github.com/rubjo/victor-mono/archive/v${version}.zip";
```

File: pkgs/data/fonts/victor-mono/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{ lib, fetchzip }:
+
+let
+  version = "1.2.1";
```
Comment (jonringer) [inline pkgs/data/fonts/victor-mono/default.nix]: feel like version could be moved below, it's already a recursive set.

**Review by jonringer — COMMENTED** [review]
Just some nitpicks :)

File: pkgs/data/fonts/victor-mono/default.nix (hunk 2 above)
Comment (dtzWill) [reply pkgs/data/fonts/victor-mono/default.nix]: Wargarbl, I lost this bouncing between whether I could/should make this fixed-output or not--but yes I'll fix those right away, sorry about that xD :).

Comment (dtzWill) [reply pkgs/data/fonts/victor-mono/default.nix]: Unfortunately appears fetchzip doesn't allow ("expect") the "version" attribute which is why it's hoisted above.  Unsurprisingly it doesn't take `pname` either :).

Comment (jonringer) [reply pkgs/data/fonts/victor-mono/default.nix]: ah, disregard then. :)

Comment (dtzWill) [reply pkgs/data/fonts/victor-mono/default.nix]: Agreed, and a fetchzipWithExtras or something seems unsatisfying as well. At least for sole purpose of avoiding the `let version = ""; in` pattern used as the alternative.  Whatcha gonna do.

Anyway, look good to folks? The double-extract is for the sake of starting from a versioned tarball :).

**Comment by worldofpeace** [comment]
It seems I've merged the competing PR, seems that this one installed `.otf` too.

Too late though.

**Comment by dtzWill** [comment]
Yes, is there a reason to prefer one over the other?
Or to "only" include one, if upstream provides both? (space?)

Never sure which way to go with these :).

---

Oh, I see--I'll take a look and maybe submit a PR with the difference
:).

**Comment by worldofpeace** [comment]
I'd distribute both.

**Comment by dtzWill** [comment]
Updated, hopefully "best of both"? :)

**Pull request closed without merging**